Alesia Yakutsionak/0/run.rb

```diff
@@ -0,0 +1,175 @@
+require 'rubygems'
+require 'bundler/setup'
+
+Bundler.require(:default)
+
+class ProductPrice
+  LAST_PRICE_DATE = '2018/10'.freeze
+
+  attr_accessor :min_price, :max_price, :last_price, :min_price_date, :max_price_date
+
+  def initialize(price, date)
+    @min_price = price
+    @min_price_date = date
+    @max_price = price
+    @max_price_date = date
+    @last_price = price if date == LAST_PRICE_DATE
+  end
+
+  def update(price, date)
+    @last_price = price if date == LAST_PRICE_DATE
+    if min_price > price
+      @min_price = price
+      @min_price_date = date
+    end
+    return unless max_price < price
+    @max_price = price
+    @max_price_date = date
+  end
+end
+
+class PriceCollector
+  MONTHS = %w(январь февраль март апрель май июнь июль август сентябрь октябрь ноябрь декабрь).freeze
+  SHEET_NUM = 0
+  DATE_ROW = 3
+  FIRST_DATA_ROW = 8
+  PRODUCT_NAME_COL = 0
+  MINSK_PRICE_COL = 6
+
+  def self.collect_data(&block)
+    print 'Please wait'
+    Dir['data/*'].each do |f|
+      if f.end_with?('.xls')
+        collect_from_xls(f, block)
```

Comment:
There is Roo::Spreadsheet with unified interface, so you don't even need to check file extensions and two different methods for parsing